util/src/lib.rs

```diff
@@ -140,3 +141,17 @@ impl<T> VecExt<T> for alloc::vec::Vec<T> {
         self.last_mut().unwrap()
     }
 }
+
+pub fn transpose_vec<T>(v: Vec<Vec<T>>) -> Vec<Vec<T>> {
+    assert!(!v.is_empty());
+    let len = v[0].len();
+    let mut iters: Vec<_> = v.into_iter().map(|n| n.into_iter()).collect();
+    (0..len)
+        .map(|_| {
+            iters
+                .iter_mut()
+                .map(|n| n.next().unwrap())
+                .collect::<Vec<T>>()
+        })
+        .collect()
```

Comment:
This seems slightly over complicated. Mostly a nit but the following might be cleaner?
```suggestion
    let outer_len = v.len();
    let inner_len = v[0].len();
    (0..inner_len)
        .map(|i| {
            (0..outer_len)
                .map(|j| v[j][i])
                .collect::<Vec<T>>()
        })
        .collect()
```